### PR TITLE
[Merged by Bors] - Bump post to v0.12.2 and post-rs to v0.7.2

### PR DESCRIPTION
**Pull request by poszu**
## Description

Bump post and post-rs to their newest versions.
Had to replace the Post Setup Manager mock with a real one because the new service version requires the postdata_metadata.json to exist when it starts.

## Test Plan

existing tests pass

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [ ] Update documentation as needed
- [ ] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

File: activation/post_supervisor.go [hunk 1]
```diff
@@ -233,7 +233,6 @@ func (ps *PostSupervisor) runCmd(
 
 		"--min-num-units", strconv.FormatUint(uint64(postCfg.MinNumUnits), 10),
 		"--max-num-units", strconv.FormatUint(uint64(postCfg.MaxNumUnits), 10),
-		"--labels-per-unit", strconv.FormatUint(postCfg.LabelsPerUnit, 10),
```
Comment (fasmat) [inline activation/post_supervisor.go:236]: This is needed for quite a lot of tests... among others all system tests. Do all nodes now need to initialize at least 64 GiB of PoST data for a test to start?

**Review by dshulyak — APPROVED** [review]

**Comment by fasmat** [comment]
I don't think this will work for testing at all, with that change a lot of tests in the `activation` and `activation/e2e` and `grpcserver` packages would need to be rewritten to not need initialized data; additionally system tests will take significantly longer if a node now has to initialize 64 GiB at minimum instead of a few KiB.

File: activation/post_supervisor.go (hunk 1 above)
Comment (poszu) [reply activation/post_supervisor.go:236]: No, the service now reads the labelsPerUnit property from the metadata JSON file. I removed this CLI argument as that information is already present when the service starts.

Comment (fasmat) [reply activation/post_supervisor.go:236]: OK I see, but how would initialization work now during system tests?

Comment (poszu) [reply activation/post_supervisor.go:236]: I don't understand. Nothing changed there - the tests should initialize the same amount of data as they used to.

Comment (fasmat) [reply activation/post_supervisor.go:236]: Wait, nvm the config option isn't removed from the node, so it still initializes with the correct labelsPerUnit 🤦‍♂️

**Comment by poszu** [comment]
Bors merge
